spec/app/domain/authentication/authn-azure/validate_application_identity_spec.rb

```diff
@@ -0,0 +1,454 @@
+
+    allow(mismatched_resource_group_annotation).to receive(:values)
+                                                     .and_return(mismatched_resource_group_annotation)
+    allow(mismatched_resource_group_annotation).to receive(:[])
+                                                     .with(:name)
+                                                     .and_return("authn-azure/resource-group")
+    allow(mismatched_resource_group_annotation).to receive(:[])
+                                                     .with(:value)
+                                                     .and_return("mismatched-resource-group")
+
+    allow(mismatched_system_assigned_identity_annotation).to receive(:values)
+                                                               .and_return(mismatched_system_assigned_identity_annotation)
+    allow(mismatched_system_assigned_identity_annotation).to receive(:[])
+                                                               .with(:name)
+                                                               .and_return("authn-azure/system-assigned-identity")
+    allow(mismatched_system_assigned_identity_annotation).to receive(:[])
+                                                               .with(:value)
+                                                               .and_return("mismatched-system-assigned-identity")
+
+    allow(mismatched_user_assigned_identity_annotation).to receive(:values)
+                                                             .and_return(mismatched_user_assigned_identity_annotation)
+    allow(mismatched_user_assigned_identity_annotation).to receive(:[])
+                                                             .with(:name)
+                                                             .and_return("authn-azure/user-assigned-identity")
+    allow(mismatched_user_assigned_identity_annotation).to receive(:[])
+                                                             .with(:value)
+                                                             .and_return("mismatched-user-assigned-identity")
+
+    allow(invalid_azure_annotation).to receive(:values)
+                                         .and_return(subscription_id_annotation)
+    allow(invalid_azure_annotation).to receive(:[])
+                                         .with(:name)
+                                         .and_return("authn-azure/test")
+
+    allow(validate_azure_annotations).to receive(:call)
+                                           .and_return(true)
+
+    allow(resource_class).to receive(:[])
+                               .with("#{account}:host:#{hostname}")
+                               .and_return(mocked_host)
+  end
+
+  #  ____  _   _  ____    ____  ____  ___  ____  ___
+  # (_  _)( )_( )( ___)  (_  _)( ___)/ __)(_  _)/ __)
+  #   )(   ) _ (  )__)     )(   )__) \__ \  )(  \__ \
+  #  (__) (_) (_)(____)   (__) (____)(___/ (__) (___/
```

Comment:
Missing tests:
1. xms_mirid that does not start with `/`
2. `RoleNotFound` raised if the role is not present (how can you mock this?)

Reply:
Added 1️⃣ 
Will look into 2️⃣

Reply:
RolenotFound, if role-id is non-existent. In other words:
```
allow(resource_class).to receive(:[])
                           .with("#{account}:host:non-existent-host")
                            .and_return(nil)
```

Reply:
Great!!! You can also omit the `.and_return(nil)` as it will happen implicitly but if you feel that writing it explicitly enhances readability then go with it.